### DISCUSSION TITLE
Run without Port 25

### DISCUSSION THREAD

**Discussion by MedicBehindYou** [docker-mailserver/docker-mailserver#3213] [Q&A] [answered]
Other

## Description

I couldn't find an answer in the documentation, though I could have just missed it, but is there a way to configure sending purely through ESMTP and only receive on ESMTP? I am teaching myself DNS, MX, DMARC etc and am using this a learning project so if this isn't possible, I apologize for wasting your time.

**Comment by georglauterbach** [comment]
Running a mail server like this, from a **technical point of view**, is a non-issue. **But**, other mail servers that start out on port 25 will not be able to connect. While modern mail servers _should_ only use secure ports (in a best-case scenario, port 465 (`submissions`)), this is sadly not always the case.

DMS will automatically upgrade connections on port 25 via STARTTLS (so it's effectively like 587). In my logs, I do not see connections on port 25 at all, but that is not to say do not exist - old mail servers might still try port 25.

**Reply by MedicBehindYou** [reply]
So when I attempt to send mail, it seems to default to Port 25 when sending to a gmail address. I have my mail client setup to only send and receive on no 25 ports so would this be an issue with my config somewhere? Here is the log (https://pastebin.com/beZdVCYC) with personal information redacted. Here is my .env file (https://pastebin.com/1aLLUckg) and my compose file (https://pastebin.com/3j45BAAJ). I have DMARC and DKIM records setup and working, but no SPF yet. I am also using cloudflare as my DNS provider if that makes any difference. Thanks for any advice you can provide!

**Reply by RebelSoftware** [reply]
port 465 should not be used. It was deprecated and then re-instated due the confusion it caused. Port 587 is the preferred port. Port 465 "if you must" 

I only use port 587 for client connections as many ISP's block port 25 to prevent users running their own mail servers.

This page has a good article on it
https://sendgrid.com/blog/whats-the-difference-between-ports-465-and-587/

**Reply by georglauterbach** [reply]
> So when I attempt to send mail, it seems to default to Port 25 when sending to a gmail address.

Yeah, this is expected. To make it short: you will likely not get around port 25 in a default scenario. **I highly recommend reading [our documentation article about ports](https://docker-mailserver.github.io/docker-mailserver/latest/introduction/#about-security-ports), because it explains everything you asked.**

**Reply by georglauterbach** [reply]
> port 465 should not be used. It was deprecated and then re-instated due the confusion it caused. Port 587 is the preferred port. Port 465 "if you must"

This is a very confusing and (except for one statement) **absolutely wrong answer!** The only true fact is:

> It was deprecated and then re-instated

It was not re-instated with the same purpose though! I recommend you also read [our documentation article about ports](https://docker-mailserver.github.io/docker-mailserver/latest/introduction/#about-security-ports) to understand that 465 is today's preferred port!

465 is for implicit TLS, i.e. TLS has to be used. 587 is explicit TLS and requires STARTTLS (which is correctly configured with DMS, but in case you misconfigure it, you will miss out on TLS with port 587). With DMS, 587 is no worse because of DMS' configuration, but what is inherently worse with port 587 is that you _could_ have non-TLS-protected traffic; with 465 you cannot.

> I only use port 587 for client connections as

Then you're missing out on implicit TLS.

> many ISP's block port 25 to prevent users running their own mail servers.

True! I heard AWS does this, for example.

> This page has a good article on it https://sendgrid.com/blog/whats-the-difference-between-ports-465-and-587/

I am not sure how Sendgrid could get this wrong, but they did 😆 You will always want to prefer implicit over explicit TLS.. I am not sure how many times we had this discussion in issues, PRs and elsewhere. Please just read our documentation for [submissions](https://docker-mailserver.github.io/docker-mailserver/latest/introduction/#outward-submission). It also cites the original [RFC8314](https://www.rfc-editor.org/rfc/rfc8314.html) accepted by the IETF. Our docs also explain why with DMS, 587 is absolutely okay to use as well.

---

To summarize, please read our docs! And you will likely want to open both ports, 465 and 587.

**Reply by RebelSoftware** [reply]
While your docs are fantastic when it comes to implementing docker-mailserver and I congratulate you on probably some of the best open source documents there are. Its not the reference guide to SMTP, rather it is the reference guide to docker-mailserver.

I do always open port 465 because most email clients continue to support it. however it is not the "industry preferred" port for smpts submission, 587 is.

You raised the [RFC8314](https://www.rfc-editor.org/rfc/rfc8314.html). Even it is at best ambiguous. It states 587 does not replace 465 (agreed) but points out the confusion around it as another service has been assigned to 465. it finishes up by saying "Encouraging the use of STARTTLS on port 587 would not create interoperability problems" You seem to be operating at odds to that statement by encouraging 465 over 587 by stating in your docs
"The best practice as of 2020 when it comes to securing Outward Submission is to use Implicit TLS connection via ESMTP on port 465 (see [RFC 8314](https://tools.ietf.org/html/rfc8314))"

Here are some more articles at odds to your documentation...

https://www.mailersend.com/blog/smtp-ports#port-465

This article is a little more "forceful"
https://www.mailgun.com/blog/email/which-smtp-port-understanding-ports-25-465-587/#subchapter-3

This one from the same source says 587 should be the default
https://www.mailgun.com/blog/email/which-smtp-port-understanding-ports-25-465-587/#subchapter-5

Here is another one confirming what the industry thinks of port 465
https://mailtrap.io/blog/smtp-ports-25-465-587-used-for/#Port-587

So while I 100% agree you should implement implicit TLS over port 465 at all times. I respectfully disagree that it should be considered the default or preferred port.

**Reply by RebelSoftware** [reply]
Just an additional note. There is no way to implement implicit TLS on port 25 the default port for all MTA to MTA communication, the STARTTLS command has become the defacto method to implement TLS on that port. However some articles discuss implementations of STARTTLS as having security issues and therefore 465 should be the default. This is rendered moot by the fact port 25 is the default port for SMTP.

The potential for security problems is a per vendor issue not a protocol issue and really is just muddying the waters.

**Reply by MedicBehindYou** [reply]
Not to commandeer this discussion, but did I have any misconfigurations that would cause STARTTLS to not work? I have both 587 AND 465 opened on the container and forwarding enabled on my router. If there is instructions on how to do this in your documentation, I must have missed it. I did include all my configs above if either of you has a moment to review them. I appreciate all the help you have provided already though!

**Reply by georglauterbach** [reply]
> Not to commandeer this discussion, but did I have any misconfigurations that would cause STARTTLS to not work? I have both 587 AND 465 opened on the container and forwarding enabled on my router. If there is instructions on how to do this in your documentation, I must have missed it. I did include all my configs above if either of you has a moment to review them. I appreciate all the help you have provided already though!

It's alright, we stole this discussion thread :D After having a quick look in your log, you could try disabling IPv6 or just set `DOVECOT_INET_PROTOCOLS=ipv4` & `POSTFIX_INET_PROTOCOLS=ipv4` as it looks like the ports cannot be bound to an IPv6 address (for whatever reason).

You also say your mail server is behind a router? I assume you made sure the router does not employ a NAT to preseve origin IP addresses.

But to be honest: you'll be much easier off using e.g. a VPS with a public IPv4 address without a NAT and an open port 25 hosted in a data center - maybe even just for starting off. After having a working instance there, you can start to alter the config to deploy your mail server in a more complex scenario. After all, mail servers are complex constructions with many moving parts.. (sadly).

**Reply by georglauterbach** [reply]
@RebelSoftware I agree, our docs are not a reference guide to SMTP and it's encryption :) You may take everything I say as something we tell users of DMS as we believe this to be in their best interest.

> I do always open port 465 because most email clients continue to support it. however it is not the "industry preferred" port for smpts submission, 587 is.

I am not sure, but maybe there is a misconception here? You seem to think 465 is legacy, and "clients continue to support it", but as a matter of fact, it is the _newer_ standard and hence, 587 is the "old" port. I am not sure about the industry-standard statement, it may be true, but I'd like to remind everyone that only because something may be an industry-standard, it does not have to be the best solution out there.

> You raised the [RFC8314](https://www.rfc-editor.org/rfc/rfc8314.html). Even it is at best ambiguous. It states 587 does not replace 465 (agreed) but points out the confusion around it as another service has been assigned to 465. it finishes up by saying "Encouraging the use of STARTTLS on port 587 would not create interoperability problems" You seem to be operating at odds to that statement by encouraging 465 over 587 by stating in your docs

Not quite; the statement is in no opposition to our docs - it just reminds you that on 587, you need to explicitly state you start TLS now (with STARTTLS) while you do not need this with 465 - that's all.

---

Some of the articles you posted seem to still be operating on the belief that 465 is SMTPS while this is technically a wrong name - the correct name by now (and already in Debian's upstream config of Postfix) is _submissions_. I agree though that there is still a lot of confusion out there. Most articles have not yet updated to the changes of the 2018 RFC it seems. They seem to be getting the technical background  wrong and only state that the intial assignement and the switch later is confusion (I agree with that though).

> So while I 100% agree you should implement implicit TLS over port 465 at all times. I respectfully disagree that it should be considered the default or preferred port.

May I ask why? To be honest, this is more a rhetorical question that everything else, because I'd like to explain again why it should be the default and the preferred port. **With that being said, we do not want to force users of DMS to use one specific port - you may use what you deem to be best for you**. I hope my statements are not too offensive, but we had this discussion a long time ago and I should probably we writing this in our docs, not here..

---

So here is a very brief explanation of why 465 is - **from a technical & security perspective** - the best port:

- port 25 is a plaintext port; we can all agree this is unacceptable as of today
- port 587 is a well-known port that starts unencrypted but by issuing the `STARTTLS` command, new  traffic will encrypted (afterwards); this is sub-optimal though, because in a wrong configuration, TLS is not enforced!
- port 465 (as of today!) is a not-too-well-known port that offers implicit TLS, i.e. traffic will **always be encrypted, no matter what**. This is - at least we as the maintainers of DMS think this way - the best option for everyone.

Please do not get me wrong, but with all due respect, I have to doubt what the articles say about the situation now - not because the opinion of others (like yours) is not important to me, but because I as a maintainer have to evaluate how something is working on a technical level and not how people think how something is working. These articles are written in good faith, no doubt - but they miss the technical background completely.

Now, as the last thing to close this off, in case you're still yearning for more, I'd like to refer you to the Postfix configuration (a sadistic ending, for sure). Have a look at [`smtpd_tls_security_level`](https://www.postfix.org/postconf.5.html#smtpd_tls_security_level). On 587 (the port you're using), the level set is `may` (!), i.e. "do not require that clients use TLS encryption". With 465, it is `encrypt` (!), i.e. "Mandatory TLS encryption".

--

I already put in way more time in this thread that I have and I am very sorry @MedicBehindYou for posting this below your question when this is not really connected to your question. @RebelSoftware if you still feel uncomfortable, I'd love you to open an issue about this, because I do not want to proceed in this discussion. I will try to update our documentation to make things even clearer. I hope I settled some of the misconceptions about the ports though. Have a nice day - I will link you on the upcoming PRs so you can have a look and maybe even post a review :) ❤️

**Reply by MedicBehindYou** [reply]
@georglauterbach Thanks so much, I’ll give those settings a try, and if I can’t get it working, I’ll take your advice and try a hosting service

**Reply by MedicBehindYou** [reply]
@georglauterbach (New logs https://pastebin.com/Aw70nGxv) So I did change the ipv4 settings for postfix and dovecot, I'm no loger failing to connect on IPV6 (now disabled), just failing on IPV4. First question: is there a specific thing I need to do to enable connecting on implicit ESMTP?  Any outgoing email only attempts to send on port 25. Does that mean that port 25 is just required for email or is my config wrong? Next question is about NAT. I do have NAT on my home router, is there something else I need to do or does that just make this a no go in general? I see this error in the logs (Apr  4 19:29:38 mail postfix/trivial-rewrite[1180]: warning: do not list domain mail.DOMAIN.com in BOTH mydestination and virtual_mailbox_domains), and from the FAQ, this appears when your FQDN is a bare domain without a prefix. In my case, I am using mail.EXAMPLE.com and not just EXAMPLE.com so I am not sure why this is showing up. I know that's quite a few questions, and in no way do I expect someone who is doing this for free to walk me through every problem I have due to my lack of experience. At the end of the day, I'm just grateful that you have made something like this in the first place. Thanks for the time reading this!

**Reply by polarathene** [reply]
@RebelSoftware I am the one who did extensive research into the ports situation back in 2020, prior maintainers of DMS also thought port 465 was no longer relevant due to it's prior deprecated assignment for SMTPS.

## SaaS supported ports

This is related to the next section, but helps illustrate the problem with advice from these vendors:

https://github.com/docker-mailserver/docker-mailserver/blob/637d27efc72e11b50da69b262943d3ca29aea14f/target/scripts/helpers/relay.sh#L309-L359

I've also since dug up my original discussions on this project from 2020 about the topic:
- https://github.com/docker-mailserver/docker-mailserver/issues/1428#issuecomment-618840753
- https://github.com/docker-mailserver/docker-mailserver/pull/1124#issuecomment-619508564

The prior Github Wiki was then updated with that information, and in 2021 I migrated that wiki to what is now our docs site.

I know I've got old notes on my system somewhere from that research with a tonne of resources / info on the topic. Port 465 is 100% the better choice where possible, but if you want to use 587 go for it :man_shrugging: 

## 465 as the submissions service

You're often citing sources that are mail focused SaaS or similar products / services.
- These tend to offer various non-standard alternative ports, and some offer port 25 but requiring authentication.
- The articles aren't always written by people that know what they're talking about. It can be someone from marketing that has enough technical understanding to research the topic and parrot the same information, or relay what the IT / engineering department communicates, etc.
- Many of the articles IIRC were originally written pre-2018 and may not have been updated to accommodate.

Instead:
- Look at what actual mail software does, like Postfix which is fairly dominant in the industry. [Port 465 (aka `submissions`)](https://github.com/vdukhovni/postfix/blob/62931e5b1f9f1e80d02a496c7fd0062a5aae1d25/postfix/conf/master.cf#L38-L41).
- `grep 465 /etc/services` on a distro will show what they consider the port. Sadly on Fedora 37 this is `smtps`, but on suse, debian/ubuntu, alpine, etc you'll get the newer 2018 `submissions` assignment. IANA is the proper source for that `/etc/services` content, where you'll find [465 is assigned for `submissions`](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=465).

Port 587 is preferred because it's been available for the purpose as the `submission` service (_you can also grep that_) for longer, and to those less familiar it avoids any confusion.

It's not preferred for being better (_plaintext connection to request an upgrade to TLS vs 465 using TLS from the start_), it requires extra work to support via reverse-proxy services as well due to STARTTLS negotiation and some MUA would leak credentials before TLS was established through 587.

---

## RFC 8314 is not ambiguous - Clearly prefers 465 deployment

> You raised the [RFC8314](https://www.rfc-editor.org/rfc/rfc8314.html). Even it is at best ambiguous.

From [RFC 8314](https://www.rfc-editor.org/rfc/rfc8314.html):

> _The STARTTLS mechanism on **port 587 is relatively widely deployed due to the situation with port 465** (discussed in [Section 7.3](https://www.rfc-editor.org/rfc/rfc8314.html#section-7.3))._
> _This differs from IMAP and POP services **where Implicit TLS is more widely deployed on servers than STARTTLS**._
> 
> _**It is desirable to migrate core protocols** used by MUA software to Implicit TLS over time, for consistency as well as for the additional reasons discussed in [Appendix A](https://www.rfc-editor.org/rfc/rfc8314.html#appendix-A)._
> _However, to maximize the use of encryption for submission, it is desirable to support both mechanisms for Message Submission over TLS **for a transition period of several years**._
> 
> _Although STARTTLS on port 587 has been deployed, **it has not replaced the deployed use** of Implicit TLS submission on port 465._

I don't see how that is ambiguous? It references the last RFC update for port 587 (in 2011). The intent of migrating to 465 for implicit TLS is very clear, but it takes a long time for this sort of change to be common place (_especially with mail-servers_), we're doing out part.

TLS 1.2 had been finalized in 2008, yet I recall support issues with Android and Mac mail clients (_Apple Mail was an issue until 2016-2018 IIRC_), mail servers themselves were also slow to upgrade (_I recall some in 2014 or so being far behind and only making the change due to SSL vulnerabilities_).

For the mail services used today like you cited, there is minimal incentive to switch something that works well enough for them and their users, especially if it were to remove support for port 587 since that'd impact many customers. You may find 465 is offered properly for `submissions` use, and eventually encouraged.

TLS 1.0 and 1.1 only recently got deprecated, and you'll probably find it takes a while until the same happens for TLS 1.2, despite 1.3 being finalized back in 2018 too.

---

## Port 25 - Bane of not breaking legacy

> There is no way to implement implicit TLS on port 25 the default port for all MTA to MTA communication, the STARTTLS command has become the defacto method to implement TLS on that port.

You can use implicit TLS on port 25, but it's against the RFC advice. It's fine for private use, just not compatible with public.

It's not because we should keep using STARTTLS, it's because that's what we had back in 1998 and to change that now causes a compatibility problem. Thus mail is permitted to be delivered in plain-text without any TLS. To try improve that we get workarounds like MTA-STS.

> However some articles discuss implementations of STARTTLS as having security issues and therefore 465 should be the default.

Port 465 was never officially meant to be a thing AFAIK?

There was a draft RFC in 1997 for it IIRC, and some mail client implemented support before STARTTLS RFC was published in 1998? I'm probably a bit off here in details, it may have been the same RFC.

I recall the STARTTLS author briefly mentioned SMTPS with 465 and got IANA to assign it, then after some revisions to [the draft revoked it](https://web.archive.org/web/20150603202057/http://www.imc.org/ietf-apps-tls/mail-archive/msg00204.html)? :man_shrugging: (_here's an [old discussion talking about it](https://narkive.com/F3ACwg2F.21)_)

> This is rendered moot by the fact port 25 is the default port for SMTP.

IIRC, that's why SMTPS didn't make sense due to the separate port it was better suited for submissions use. Non-issue for STARTTLS though which was a useful improvement for port 25.

I am doubtful we'd see much change here as it's likely too invasive.

**Reply by polarathene** [reply, marked as answer]
@MedicBehindYou 

> First question: is there a specific thing I need to do to enable connecting on implicit ESMTP?
> Any outgoing email only attempts to send on port 25.
> Does that mean that port 25 is just required for email or is my config wrong?

If you're sending mail directly to another address, you'd do this through port 25. Just like when you receive mail from another mail server.

If you are relaying your mail through another host like SendGrid, or Gmail to forward your outgoing mail from their mail servers, you will typically connect to a submission(s) port (587 or 465). These require credentials to authenticate with that third-party service, we support this via relay config (_which needs to be refactored at some point_).

You'll want to use port 587 in this case at present, or a use `postfix-main.cf` /  `user-patches.sh` to configure for port 465 (_DMS presently lacks a convenient switch between 587 and 465_). This is described here:

- https://github.com/docker-mailserver/docker-mailserver/issues/2601#issuecomment-1137016558
- https://github.com/docker-mailserver/docker-mailserver/pull/2607

DMS can receive inbound mail submission on ports 587 or 465 though with no extra config changes necessary. This is for mail clients like ThunderBird, but could also be other mail servers (like another DMS instance) that relays mail through your server.

---

> Next question is about NAT.
> I do have NAT on my home router, is there something else I need to do or does that just make this a no go in general?
> I see this error in the logs:
> ```
> Apr 4 19:29:38 mail postfix/trivial-rewrite[1180]: warning: do not list domain mail.DOMAIN.com in BOTH mydestination and virtual_mailbox_domains
> ```
> from the FAQ, this appears when your FQDN is a bare domain without a prefix.
> In my case, I am using `mail.EXAMPLE.com` and not just `EXAMPLE.com`

I am not sure about NAT, another maintainer or user might be able to comment on that.

The warning log should be irrelevant to that. Something else in your setup is probably related to that.
- Avoid using host `network` for the container if you can, as that will introduce some issues. The default should be the docker bridge.
- You should have no `domainname:` in your config, just `hostname: mail.example.com`.
- When adding an account, it would still be `user@example.com`, not `user@mail.example.com`.
- If you're still having an issue, try with `:edge` to confirm if it's not been fixed with upcoming v12 release next week.
- Make sure you are re-creating the container from scratch `docker-compose up --force-recreate` (_if you're stopping it with `ctrl + c`_) or `docker-compose down` before `docker-compose up`.
- Also try without any volumes mounted. Create an account: `docker exec -it container_name_here setup email add hello@example.com` and check the logs to see if you still have the mentioned warning log (_may require mail receive/send to trigger_).

If none of that resolves it, raise a proper bug report on our issue tracker with more information :+1: 

---

> I know that's quite a few questions, and in no way do I expect someone who is doing this for free to walk me through every problem I have due to my lack of experience.
> At the end of the day, I'm just grateful that you have made something like this in the first place.
> Thanks for the time reading this!

No worries! Thanks for the appreciation :heart: 

If you do find anything tricky like the above and you feel it's probably something other new users could stumble with, we always appreciate contributions to the docs :+1:

**Reply by georglauterbach** [reply]
@MedicBehindYou the answer @polarathene has given is really good, try to follow his advice :)

About the NAT: simply using it is definitely a no-go! My mail server for example would reject such e-mails. As the IP of your server "becomes" the outbound IP of your router due to the NAT, the `A` and `PTR` records do not match anymore, in fact, just resolving the DNS name (with the help of the `A` record) will produce a mismatch. It is possible to preserve the original IP, but more complex. I'd therefore advise, maybe just for your initial months, trying a hosting provider to give you a very easy setup :)

**Reply by MedicBehindYou** [reply]
@polarathene @georglauterbach Thank you so much for your time and help! I've been rather busy but I'll give these suggestions a try soon.

**Comment by polarathene** [comment]
> I am teaching myself DNS, MX, DMARC etc and am using this a learning project

This isn't a complete guide, but [I did share config for running a local DNS + `docker-mailserver` container(s)](https://github.com/docker-mailserver/docker-mailserver/issues/2601#issuecomment-1135520225), it can run completely offline.

You can setup the TLS certs for an offline test environment to experiment with between multiple DMS containers too. I have [documented how we create the needed cert files](https://github.com/docker-mailserver/docker-mailserver/blob/637d27efc72e11b50da69b262943d3ca29aea14f/test/test-files/ssl/example.test/README.md).

Might be helpful for you :+1:

**Reply by Jarrodsz** [reply]
Im also on digitalocean which blocks port 25.

How many % of mail clients would not try to connect to port 587 when port 25 is blocked. Are there some statistics on that?

Having my own mailserver the flexibility and not paying several eur each month to host mails at a provider seems to be more important then get some issues with people not be able to contact me, but then im still curious if, and how many times one vould expect mails to nog get delivered cause you dont have a port 25.

Anyone has expierience with this?

I guess all modern email clients supprt port 587 delivery and all mayor online mail providers support it gmail outlook protonmail etc?

Is my assumption corect on this?

I would say shame for
Someone still using windows xp and windows mail if it would not support port 587.

It would still s*** though if your missing important emails due to this but the benefits of own mailserver outweight this.

I know you can relay with things like sendgrid for outgoing mail perhaps there is a free service you can use to " forward port 25 " to 587 true dns but guess not the port is open or closed on the mailserver ip and thats the end of it i guess

**Reply by polarathene** [reply]
> Im also on digitalocean which blocks port 25.

Usually port 25 outbound is blocked, preventing sending mail, but not preventing receiving. You should of course check that mail can be received on 25 as it's possible inbound traffic is also blocked, though I don't know why it would be.

> How many % of mail clients would not try to connect to port 587 when port 25 is blocked. Are there some statistics on that?

You should be submitting mail via port 587 or 465, not 25. These two ports require authentication and ensure TLS (_to the mail server, not necessarily for delivery from it to another_).

25 is typically for mail servers to send mail out on to be received by port 25 on another mail server.

> how many times one vould expect mails to nog get delivered cause you dont have a port 25.

From any mail account not managed by your mail server, that would be highly likely the majority of mail. If any third-party is to send mail to you on port 587 / 465, they must have credentials to login. Untrusted third-parties won't have that, they must go through port 25.

> Someone still using windows xp and windows mail if it would not support port 587.

Pretty sure if anyone was on Windows XP they'd have problems establishing TLS as they're lacking modern cipher suites, so even if they connected to port 587 it should be unsuccessful at securing the connection thus rejected.

> I know you can relay with things like sendgrid for outgoing mail perhaps there is a free service you can use to " forward port 25 " to 587 true dns but guess not the port is open or closed on the mailserver ip and thats the end of it i guess

Yes, same approach. A mail relay that can receive mail for a domain (MX record points to it) would then be configured to send that mail to another mail server via port 587/465 (an account for the relay). You should be able to use DMS to do this at both ends, but need to ensure that it's only relaying mail destined to your domain(s), I think Postfix offers "relay domains" for this, but DMS does not have any support implemented for that feature.

**Reply by Jarrodsz** [reply]
> > Im also on digitalocean which blocks port 25.
> 
> 
> 
> Usually port 25 outbound is blocked, preventing sending mail, but not preventing receiving. You should of course check that mail can be received on 25 as it's possible inbound traffic is also blocked, though I don't know why it would be.

Ah right my misconception so i do enable port 25 then in my docker compose file. 
> 
> 
> 
> > How many % of mail clients would not try to connect to port 587 when port 25 is blocked. Are there some statistics on that?
> 
> 
> 
> You should be submitting mail via port 587 or 465, not 25. These two ports require authentication and ensure TLS (_to the mail server, not necessarily for delivery from it to another_).
> 
> 
> 
> 25 is typically for mail servers to send mail out on to be received by port 25 on another mail server.
> 

SMTP port 25 is blocked on all Droplets for new accounts to prevent spam and other abuses of our platform. SendGrid delivers your transactional and marketing emails through the world's largest cloud-based email delivery platform.

It seems this is only for outgoing mail which is fine although its not clearly states if digitalocean also blocks incoming 25. I guess recieve mail is not an issue for them but outgoing mail on port 25 is. My assumption was port 25 would also be used for sending but this was my bad. Port 587 with ssl is for sending instead ( alternative) so its a non issue then.


> 
> > how many times one vould expect mails to nog get delivered cause you dont have a port 25.
> 
> 
> 
> From any mail account not managed by your mail server, that would be highly likely the majority of mail. If any third-party is to send mail to you on port 587 / 465, they must have credentials to login. Untrusted third-parties won't have that, they must go through port 25.
> 
> 
> 
> > Someone still using windows xp and windows mail if it would not support port 587.
> 
> 
> 
> Pretty sure if anyone was on Windows XP they'd have problems establishing TLS as they're lacking modern cipher suites, so even if they connected to port 587 it should be unsuccessful at securing the connection thus rejected.
> 
> 
> 
> > I know you can relay with things like sendgrid for outgoing mail perhaps there is a free service you can use to " forward port 25 " to 587 true dns but guess not the port is open or closed on the mailserver ip and thats the end of it i guess
> 
> 
> 
> Yes, same approach. A mail relay that can receive mail for a domain (MX record points to it) would then be configured to send that mail to another mail server via port 587/465 (an account for the relay). You should be able to use DMS to do this at both ends, but need to ensure that it's only relaying mail destined to your domain(s), I think Postfix offers "relay domains" for this, but DMS does not have any support implemented for that feature.

**Reply by polarathene** [reply]
> My assumption was port 25 would also be used for sending but this was my bad. Port 587 with ssl is for sending instead ( alternative) so its a non issue then.

Just to clarify, ports 587 and 465 are submission ports. They are for authenticating with your mail-server (DMS) and sending it mail to deliver to an account it manages, or send outbound (_typically as the account you've logged in and sent the mail from your mail client_).

If you have setup SendGrid like DigitalOcean mentions, then you'll relay mail you send through them in a similar way. They have a variety of non-standard ports like 2525, and their port 25 behaves more like 587 as authentication is required with a SendGrid account.

You configure DMS with our relay host settings support with that SendGrid information and all your mail sent outbound will go through SendGrid (_which has better delivery reputation which better avoids arriving in spam/junk or being rejected by the mail-server your mail was sent to_).
